### ISSUE TITLE
cylc message payload

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#2502]
Messages from task jobs are currently just single strings.  Ideally users could send an arbitrary serializable payload (probably a dict) to be passed on to event handlers.  This could be particularly useful when suites or suite components may be reused in different contexts, so that the suite or event handler does not have to know additional context-specific information about the event represented by a message.

**Comment by matthewrmshin** [comment]
Sounds like we have a revival of #181. I think the solution proposed by https://github.com/cylc/cylc/issues/2214#issuecomment-341683390 should help to solve this.

**Comment by matthewrmshin** [comment]
Just to elaborate, I think the proposed CLI wrapper for the network API can be used for just this - as I expect the command to have a simple interface such as:

```bash
cylc client my-suite put_message <<<'__JSON__'
{"task_id": "mytask.1", "severity": "CUSTOM", "message": "a\ngigantic\nmessage\ntower"}
__JSON__
```

**Comment by hjoliver** [comment]
#181 was about providing additional information held by the suite server program to event handlers (and now we have quite an extensive list of possible event handler args of that kind).

This is about allowing task jobs to send arbitrary data back to the suite server program, not just a text string.  (However, your second comment, which was posted as I wrote this, shows you know that).

**Comment by matthewrmshin** [comment]
Obviously, we can also modify `cylc message` to be able to read from STDIN.

(As an aside, it may be time that we rationalise the CLI of `cylc message` to be a bit more like the other client commands. Suggest `cylc message [OPTIONS] SUITE TASK-ID MESSAGE`. If `MESSAGE` is `-`, read message from STDIN.)
